File: 0078-subsets/0078-subsets.java
class Solution {
    static void findSubset(int i,int[] nums,List<List<Integer>> result,List<Integer> subset){
        if(i == nums.length){
            result.add(new ArrayList<>(subset));
            return;
        }

        subset.add(nums[i]);
        findSubset(i+1, nums, result, subset);
        subset.remove(subset.size() -1);
        findSubset(i+1, nums, result, subset);

    }
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> result = new ArrayList<>();
        findSubset(0, nums, result, new ArrayList<>());
        return result;
    }
}

File: 0078-subsets/0078-subsets.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var subsets = function(nums) {
    var result = [];
    generateSubsets(nums,0,[],result);
    return result
};
function generateSubsets(nums,i,current,result){
    if(i == nums.length){
        result.push([...current]);
        return;
    }
    current.push(nums[i]);
    generateSubsets(nums,i+1,current,result);
    current.pop();
    generateSubsets(nums,i+1,current,result);
}